base/dict.jl

```diff
@@ -270,7 +270,11 @@ function rehash!(h::Dict{K,V}, newsz = length(h.keys)) where V where K
     return h
 end
 
-function sizehint!(d::Dict, newsz)
+max_values(::Type) = typemax(Int)
```

Comment:
I think this is safe. Such a set would probably malfunction badly with a `Dict` that has a negative underlying `count`...